File: 221007/tappytoon/2.py
def solution(n):
    print(list(map(int, list(dict.fromkeys(list(str(n)))))))
    intArr = list(map(int, list(dict.fromkeys(list(str(n))))))
    intArr2 = [i for i in intArr if i != 0]

    count = 0

    for i in intArr2:
        if n % i == 0:
            count += 1

    return count

# 0은 빼는 로직 추가


print(solution(2232))

File: 221007/야놀자/2.js
const getCount = (copiedS, word) => {
    let count = 0

    while (true) {
        let end = false;

        [...word].forEach((w, wi) => {
            const index = copiedS.findIndex((c) => c === w)
            copiedS.splice(index, 1)

            if (index === -1) {
                end = true;
            }

            if (index !== -1 && wi === [...word].length - 1) {
                count += 1;
            }
        })

        if (end) {
            break
        }
    }

    return count;
}

function solution(S, L) {
    const counts = [];

    L.forEach((ele) => {
        const count = getCount([...S], [...ele]);
        counts.push(count);
    })

    return Math.max.apply(null, counts);
}

console.log(solution('BILLOBILLOLLOBBI', ['BILL', 'BOB']));
